src/product/product.service.spec.ts

```
@@ -0,0 +1,194 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ProductService } from './product.service';
+import { PrismaService } from '../prisma/prisma.service';
+import { mapPrismaProductToDomain } from './product.mapper';
+import { CreateProductDto, UpdateProductDto } from './dto/product.dto';
+import { ProductId, CategoryId, Price, ProductName, ProductDescription, SKU, StockQuantity } from './types';
+import { NotFoundException } from '@nestjs/common';
+import { Product as ProductInterface } from './interfaces/product.interface'; // Domain Product interface
```

ESLint

[error] 'ProductInterface' is defined but never used.

(@typescript-eslint/no-unused-vars)